vital/util/numeric.h

```diff
@@ -0,0 +1,39 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+/// \file
+/// \brief Utility functions regarding arithmetic types.
+
+#include <type_traits>
+
+#include <cmath>
+
+namespace kwiver {
+
+namespace vital {
+
+#ifdef _MSC_VER
+
+// isnan() wrapper which accepts integral types, even on noncompliant MSVC.
+// See https://github.com/microsoft/STL/issues/519
+template < class T >
+bool
+isnan( T value )
+{
+  // TODO(C++17): make if constexpr and remove the static_cast
```

Comment:
Do we actually support any pre-C++17 MSVC? Since this code is MSVC-only (the main pre-C++17 compiler I know we are — for the moment — still supporting is GCC), can we go ahead and make this change? Or do we still need it to work on MSVC that doesn't know `if constexpr`?

Reply:
Not knowing the answer (or if C++17 is guaranteed to be enabled on MSVC right now even if it is supported), I would err on the side of safety.